etcdmain/etcd.go

```diff
@@ -333,35 +335,28 @@ func startProxy(cfg *config) error {
 	return nil
 }
 
-// setupCluster sets up an initial cluster definition for bootstrap or discovery.
-func setupCluster(cfg *config) (*etcdserver.Cluster, error) {
-	var cls *etcdserver.Cluster
-	var err error
+// setupPeerURLsMapAndToken sets up an initial peer URLsMap and cluster token for bootstrap or discovery.
+func setupPeerURLsMapAndToken(cfg *config) (urlsmap types.URLsMap, token string, err error) {
```

Comment:
agree. i have thoughts about moving this function into etcdserver pkg and let etcdserver do nothing about config parsing.